qadence/backends/pyqtorch/convert_ops.py

```diff
@@ -114,36 +157,47 @@ def convert_block(
 
     if isinstance(block, ScaleBlock):
         scaled_ops = convert_block(block.block, n_qubits, config)
-        scale = extract_parameter(block, config)
-        return [pyq.Scale(pyq.Sequence(scaled_ops), scale)]
+        scale = extract_parameter(block, config=config)
+
+        # replace underscore by dot when underscore is between two numbers in string
+        if isinstance(scale, str):
+            scale = re.sub(r"(?<=\d)_(?=\d)", ".", scale)
```

Comment:
This substitution is required due to the method how `qadence` stringifies `sympy` expressions. Your corner case is interesting - I have to test it, but it would seem that current implementation would substitute a dot in this variable. And that is not desirable. The reason for this regular expression is to convert floats from form `0_1234` to `0.1234` that is understood by `parse_expr` method of `sympy`. But yes, it has to be updated to handle your case

Reply:
fixed